src/AddressUkPafLookupView.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+
+namespace Gcd\UkAddressLookup;
+
+
+use Rhubarb\Leaf\Presenters\Controls\Buttons\Button;
+use Rhubarb\Leaf\Presenters\Controls\ControlView;
+use Rhubarb\Leaf\Presenters\Controls\Selection\DropDown\DropDown;
+use Rhubarb\Leaf\Presenters\Controls\Text\TextBox\TextBox;
+
+class AddressUkPafLookupView extends ControlView
+{
+    protected $htmlType = "address";
+
+    public function __construct($htmlType = "address")
+    {
+        $this->htmlType = $htmlType;
+
+        $this->requiresContainer = true;
+        $this->requiresStateInputs = true;
+    }
+
+    public function getDeploymentPackage()
+    {
+        $package = parent::getDeploymentPackage();
+        $package->resourcesToDeploy[] = __DIR__ . "/AddressUkPafLookupViewBridge.js";
+
+        return $package;
+    }
+
+    protected function getClientSideViewBridgeName()
+    {
+        return "AddressUkPafLookupViewBridge";
+    }
+
+    public function createPresenters()
+    {
+        $this->AddPresenters(
+            $country = new DropDown("Country"),
+            new TextBox("HouseNumber", 20),
+            new TextBox("PostCodeSearch", 10),
+            $search = new Button("Search", "Search", function() {}),
+            new TextBox("Line1", 50),
+            new TextBox("Line2", 30),
+            new TextBox("Town", 30),
+            new TextBox("County", 20),
+            new TextBox("PostCode", 10)
+        );
+
+        $country->SetSelectionItems( [ [ "", "Please select..." ], Country::getCountriesList() ] );
+    }
+
+    public function printViewContent()
+    {
+        $this->printFieldset( "", [ "Country" ] );
```

Comment:
My point is that the postcode search is only relevant for UK addresses. If the address is outside the UK, it should switch to manual address entry.